### PR TITLE
build(ios): Bump sentry-cocoa to 7.0.0 and remove setLogLevel

### PR DESCRIPTION
**Pull request by jennmueng**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

Bumps `sentry-cocoa` to `7.0.0-beta.0` and uses the `SentrySDK.storeEnvelope` and `SentrySDK.captureEnvelope` methods.

Related: https://github.com/getsentry/sentry-dart/pull/424

Removes the `setLogLevel` bridge method as this was removed in Cocoa.
- On iOS simply just passing `debug: true` in the options already sets the log level by default, so no action is needed there.
- Directly sets the logger level to `Logger.INFO` on Android when `debug` is true on `startWithOptions`.

## :bulb: Motivation and Context

Needs new close method added in cocoa for #1457 

## :green_heart: How did you test it?
Tested on sample app with both iOS and Android.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by jennmueng** [comment]
Moved to blocked due to naming conflict in `sentry-cocoa` with `Random.m` (See failing iOS e2e test CI) as discussed with @philipphofmann

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- build(ios): Bump sentry-cocoa to 7.0.0-beta.1 and remove setLogLevel #1459
```
Comment (bruno-garcia) [inline CHANGELOG.md:5]:
```suggestion
- build(ios): Bump sentry-cocoa to 7.0.0 and remove setLogLevel #1459
```

File: RNSentry.podspec [hunk 2]
```diff
@@ -17,7 +17,7 @@ Pod::Spec.new do |s|
   s.preserve_paths = '*.js'
 
   s.dependency 'React-Core'
-  s.dependency 'Sentry', '6.1.4'
+  s.dependency 'Sentry', '7.0.0-beta.1'
```
Comment (bruno-garcia) [inline RNSentry.podspec:20]:
```suggestion
  s.dependency 'Sentry', '7.0.0'
```

**Review by bruno-garcia — COMMENTED** [review]
7.0.0 just came out!

**Review by bruno-garcia — APPROVED** [review]

**Comment by jennmueng** [comment]
Re-running e2e test as iOS pods failed due to outdated specs.

**Pull request merged**